tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -472,4 +472,136 @@ TEST_F(
   }
 }
 
+class AGOverlapTest : public MultiDeviceTest {
+ protected:
+  OverlapTestParams params;
+
+  int64_t num_devices_;
+  int64_t my_device_index_;
+  std::vector<int64_t> all_devices_;
+  at::Tensor ta_, tb_, tc_;
+  at::Tensor ta_sharded_, tc_sharded_;
+  // stores the backend
+  c10d::Backend* world_communicator_;
+
+  // Define I/O and intermediate Tensor shapes
+  // A has shape (S, sharded(n), M/(S*n), K)
+  // B(K,N)
+  // C has shape (S, M/S, N)
```

Comment:
```suggestion
  // A has shape (S, sharded(D), M/(S*D), K)
  // B(K,N)
  // C has shape (S, M/S, N)
```
Nit: I assume `n` is the number of devices? I changed it to D to avoid confusion with the other `N`, but I'm open to other ways to avoid confusion.